internal/exec/util/file.go

```diff
@@ -37,20 +35,25 @@ const (
 )
 
 type File struct {
-	io.ReadCloser
-	hash.Hash
+	Hash        hash.Hash
 	Path        string
 	Mode        os.FileMode
 	Uid         int
 	Gid         int
 	expectedSum string
+	Url         url.URL
+	Client      *resource.HttpClient
+	Compression string
 }
 
-func (f File) Verify() error {
+// Verify, given a sum of bytes produced by a hash.Hash, will check if the bytes
```

Comment:
> If there is no hash expected for this File, nil is returned.
I still gotta pass an empty byte slice in this case to match the expectedSum empty byte slice. Seems like you can drop that line